IQKeyBoardManager/Categories/IQUIView+Hierarchy.m

```diff
@@ -198,6 +205,16 @@ - (NSArray*)deepResponderViews
     return textFields;
 }
 
+- (UIView *)parentView
+{
+    UIView *parentView = self;
+    while (![parentView.superview isKindOfClass:[UIWindow class]]) {
```

Comment:
When there will be no window for the view. I think the while loop never ends. I think you should also put a condition to check if parentView.superview != nil